src/Pim/index.tsx

```diff
@@ -235,7 +240,7 @@ class Pim extends React.PureComponent {
     this.onItemSave = this.onItemSave.bind(this);
   }
 
-  public onItemSave(item: PimType, journalUid: string, originalEvent?: PimType) {
+  public onItemSave(item: PimType, journalUid: string, originalEvent?: PimType, goBack = true) {
```

Comment:
I don't see anywhere where you used this extra parameter, nor do I understand why it's needed. Could you please elaborate?
If I had to assume, I guess you did it for marking items as "done" in the task list page without going back, though I can't find where it's used.

Additionally, rather than have this extra parameter, I think it would be cleaner to remove the history.goback altogether from this code, make it return a promise, and then whichever piece of code that needs to go back after the save is done, can just have it there (probably just in contactedit, eventedit, and taskedit). What do you think?

Reply:
I am pretty much just using this for the task "quick add" feature. Without that parameter, the app takes you back to the homepage every time you try to add a task. But what I want is to be able to stay on the same page after saving. 

I know this solution seems silly, but I did it to change the least amount of code possible. I agree that a promise would make much more sense. Would require editing some other code. I can handle that in a separate PR

Reply:
As I wrote, I understand why you did it, I just couldn't find anywhere you called this function with this extra parameter. Maybe I missed something because of all the code I had to review, but I don't remember seeing anything where it's used.

Anyhow, I think the promise is a better way of doing it, and how it should have been done in the first place. As I told you, the current code is less than awesome in some parts. This is one of them.